vowpalwabbit/io_to_queue.h

```diff
@@ -0,0 +1,76 @@
+#ifndef _IO_QUEUE_
+#define _IO_QUEUE_
+
+#include <thread>
+#include <future>
+#include <mutex>  
+#include <chrono>   
+
+#include "io_item.h"
+
+//Adds a line of input to the input queue
+inline bool add_to_queue(vw& all, char *& line){
+
+  bool finish = false;
+
+  size_t num_chars_initial = readto(*(all.p->input), line, '\n');
+  
+  if(num_chars_initial < 1 || strlen(line) < 1){
```

Comment:
readto will strip characters from whatever was read in the file. I think that can cause a situation where `strlen(line)` == 0 but we haven't finished reading the file yet